src/main/java/org/testng/xml/XmlSuite.java

```diff
@@ -79,8 +79,25 @@ public String toString() {
   }
 
   /** Configuration failure policy options */
-  public static final String SKIP = "skip";
-  public static final String CONTINUE = "continue";
+  public enum FailurePolicy {
+    SKIP, CONTINUE;
+
+    public static FailurePolicy getValidPolicy(String policy) {
+      if (policy == null) {
+        return null;
+      }
+      try {
+        return XmlSuite.FailurePolicy.valueOf(policy.toUpperCase());
+      } catch (IllegalArgumentException e) {
+        return null;
+      }
+    }
+
+    @Override
+    public String toString() {
+      return name().toLowerCase();
```

Comment:
Maybe specify the names in the constructor (`SKIP("skip")`) so that someone modifying `toString()` by accident won't break the parsing.